### PR TITLE
Move all CLI libraries to be single-TFM targeting netstandard1.3 where possible.

### PR DESCRIPTION
**Pull request by eerhardt**
Microsoft.Extensions.Testing.Abstractions is the only one I couldn't move to netstandard1.3 that should be.  But this library should be removed with the new `dotnet test` strategy.

@livarcocc @piotrpMSFT @brthor @jgoshi

### CODE REVIEW

File: src/Microsoft.DotNet.Cli.Utils/project.json [hunk 1]
```diff
@@ -15,14 +15,7 @@
...
   "frameworks": {
-    "net451": {
-      "frameworkAssemblies": {
-        "System.Runtime": {
-          "type": "build"
-        }
-      }
-    },
-    "netstandard1.6": {
+    "netstandard1.3": {
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/project.json]: Maybe you should hold on this change for this file. It probably will pick up a dependency on MSBuild and we don't know if they can support netstandard1.3. They did indicate that supporting net451 should be feasible.

File: src/Microsoft.DotNet.Cli.Utils/CommandResolution/OutputPathCommandResolver.cs [hunk 2]
```diff
@@ -80,7 +80,7 @@ private ProjectContext GetProjectContextFromDirectory(string directory, NuGetFra
             var projectContext = ProjectContext.Create(
                 projectRootPath,
                 framework,
-                DotnetRuntimeIdentifiers.InferCurrentRuntimeIdentifiers());
+                RuntimeEnvironmentRidExtensions.GetAllCandidateRuntimeIdentifiers());
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/OutputPathCommandResolver.cs]: Why is it that this does not need the RID cap from InferCurrentRuntimeIdentifiers?

Comment (eerhardt) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/OutputPathCommandResolver.cs]: 1. This code should be going away.
2. This code can't guarantee that it is running in the context of a CLI, so there is no way to get to the CLI's .version file.  Thus, no way to RID cap.

File: src/Microsoft.DotNet.Cli.Utils/project.json (hunk 1 above)
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/project.json]: I think we agreed on this going to netstandard1.5. Can you make the change here?

**Pull request merged**